### ISSUE TITLE
conditionals for tags don't work

### ISSUE THREAD

**Issue by ukleinek** [neomutt/neomutt#4402]
## Expected Behaviour

I have the following in my settings: 
```
set index_format = "%4C %Z %<GA?A&%<GS?S& >> %{%b %d} %-15.15L (%?l?%4l&%4c?) %s"
tag-formats \
        "killed" "GK" \
        "archive" "GA" \
        "inbox" "GI" \
        "spam" "GS"
```

With neomutt 20231103+dfsg1-1+b2 (as shipped by Debian), mails got marked with an "A" if they have the "archive" tag and otherwise with an "S" if they have the "spam" tag.

neomutt 20241002+dfsg-1 just shows " " for all mail, no matter if they have the "archive" or "spam" tag.

## related irc talk

```
[1728462176]   ukleinek: I'm using: set index_format = "%4C %Z .%<GA?A&%<GS?S& >>. %{%b %d} %-15.15L (%?l?%4l&%4c?) %s"
[1728462180]   ukleinek: (dots only for debugging)
[1728462236]   ukleinek: And I only see ". ." in the index view.
[1728462240]   ukleinek: tag-formats "killed" "GK" "archive" "GA" "inbox" "GI" "spam" "GS"
[1728462402]   ukleinek: %J works as expected, i.e. there are mails with the archive tag, which got an "A" in 20231103+dfsg1-1+b2, but not with 20241002+dfsg-1
...
[1728467194]    flatcap: ukleinek: hmm... there's a known problem with the format strings, but yours ok
[1728467202]        flatcap: looks ok*
...
[1728467711]   ukleinek: ..ooOO(My problem looks ok? I don't agree :-)
[1728467813]        flatcap: sorry... the problem I expected isn't present in your string
[1728467826]        flatcap: I'm trying to test yours now
[1728468025]        flatcap: it seems to work for me...
[1728468036]        flatcap: I have:
[1728468046]        flatcap: tag-transforms "inbox"   "i" 
[1728468048]        flatcap: tag-formats "inbox"   "GI"
[1728468069]        flatcap: %GI displays 'i' if the email has an 'inbox' tag
[1728468113]        flatcap: and %GI works conditionally, as you've done in your format string
[1728468271]        flatcap: do you set "tag-transforms" ?
[1728468687]        flatcap: hmm...
[1728468864]        flatcap: the "inbox" conditional works, but any other tag seems to fail
[1728469727]        flatcap: sorry, chores to do, back shortly
[1728471255] ::!!:: ethernet_zero [~eth0@user/ethernet-zero/x-5643926] has joined #neomutt
[1728472716]        flatcap: re
[1728475079]    flatcap: ukleinek: ok, I think I've found two problems
[1728475130]        flatcap: 1. `tag-transforms` seem to be _required_
[1728475187]        flatcap: 2. NeoMutt can't seem to do conditionals based on non-ASCII `tag-transforms`
[1728475312]        flatcap: I need to find some other notmuch users
[1728479632]   ukleinek: flatcap: tag-transforms is unset here.
[1728479711]   ukleinek: hmm, the manual is sparse about tag-transforms
[1728479819]        flatcap: yeah.  I _think_ your problem will be solved by setting it, even if it's trivial:  
                             tag-transforms "inbox" "inbox" "archive" "archive" ...
[1728479872]        flatcap: I'm still trying to understand the dark magic that powers %<GI?...
[1728480050]   ukleinek: tag-transforms "inbox" "inbox" "archive" "archive" "spam" "spam" <- doesn't help
[1728480153]        flatcap: hmm... ok
[1728480154]   ukleinek: %GA displays "archive" then
[1728480223]   ukleinek: %<GA?A&a> shows "a" for all mail.
[1728480248]        flatcap: ok
[1728480372]  * ukleinek downgrades neomutt and prepares building it himself in case a patch appears to test.
[1728480518]        flatcap: that may take a while :-)
[1728480577]        flatcap: if you've got a second, please can you raise an issue on github -- a quick description + sample 
                             config would be great 
[1728480645]        flatcap: (I'm debugging now, but if I can't solve it quickly, I may have to come back to it another day)
```

## How often does this happen?

- Always

## When did it start to happen?

When I upgraded from 20231103+dfsg1-1+b2 to 20241002+dfsg-1. I tried an upgrade already earlier to a version in between these two, probably 20240425+dfsg-2, and there the index was completely broken.

## NeoMutt Version

```
NeoMutt 20241002
Copyright (C) 2015-2024 Richard Russon and friends
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 6.10.9-amd64 (x86_64)
ncurses: ncurses 6.5.20240427 (compiled with 6.5.20240427)
libidn2: 2.3.7 (compiled with 2.3.7)
GPGME: 1.18.0
GnuTLS: 3.8.6
libnotmuch: 5.6.0
PCRE2: 10.42 2022-12-11
storage: tokyocabinet, lmdb
compression: lz4, zlib, zstd

Configure options: --build=x86_64-linux-gnu --prefix=/usr {--includedir=${prefix}/include} {--mandir=${prefix}/share/man} {--infodir=${prefix}/share/info} --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules {--libdir=${prefix}/lib/x86_64-linux-gnu} --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking --mandir=/usr/share/man --libexecdir=/usr/libexec --with-mailpath=/var/mail --autocrypt --disable-paths-in-cflags --full-doc --gnutls --gpgme --gsasl --gss --lmdb --lua --lz4 --notmuch --pcre2 --sqlite --tokyocabinet --zlib --zstd

Compilation CFLAGS: -g -O2 -Werror=implicit-function-declaration -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -std=c11 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -DNCURSES_WIDECHAR -isystem /usr/include/mit-krb5 -O2

Compile options:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo +gnutls +gpgme 
  +gsasl +gss +hcache -homespool +idn +inotify -locales_hack +lua +nls +notmuch 
  -openssl +pcre2 +pgp -sasl +smime +sqlite +truecolor 

MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version
  Debian testing

* Were you using multiple copies of NeoMutt at once?
  no

* Were you using 'screen' or 'tmux'?
  Yes, I use tmux, but I don't start neomutt always directly in a terminal shell :-)

* Is your email local (maildir) or remote (IMAP)?
   local mail

**Comment by flatcap** [comment]
Thanks for the bug report, @ukleinek.

I've force-pushed a better fix (commit d28d5ad71) to [[devel/tags]](https://github.com/neomutt/neomutt/tree/devel/tags).
(and given it a lot of testing)

If you have a second, please can you check this works for you too.

Thanks!

**Comment by ukleinek** [comment]
d28d5ad71eb1 ("tags: fix conditionals") works fine in my 1-second test.
